src/validation_workflow/docker/validation_docker.py

```diff
@@ -29,23 +26,56 @@ def __init__(self, args: ValidationArgs) -> None:
-            self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
-            self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
-
-            self.local_image_OS_id = self.get_image_id(self._OS_image_name, self.args.version)
-            self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, self.args.version)
+        try:
+            assert self.is_container_daemon_running(), 'Docker Daemon is not running. Exiting the docker validation.'
+
+            image_names = {
+                'dockerhub': {
+                    'os': {
+                        True: 'opensearchstaging/opensearch',
+                        False: 'opensearchproject/opensearch'
+                    },
+                    'osd': {
+                        True: 'opensearchstaging/opensearch-dashboards',
+                        False: 'opensearchproject/opensearch-dashboards'
+                    }
+                },
+                'ecr': {
+                    'os': {
+                        True: 'public.ecr.aws/opensearchstaging/opensearch',
+                        False: 'public.ecr.aws/opensearchproject/opensearch'
+                    },
+                    'osd': {
+                        True: 'public.ecr.aws/opensearchstaging/opensearch-dashboards',
+                        False: 'public.ecr.aws/opensearchproject/opensearch-dashboards'
+                    }
+                }
+            }
+
+            # Choose image names based on input arguments
+            self._OS_image_name = image_names[self.args.docker_source]['os'][self.args.using_staging_artifact_only]
+            self._OSD_image_name = image_names[self.args.docker_source]['osd'][self.args.using_staging_artifact_only]
+
+            self.local_image_OS_id = (
+                self.get_image_id(self._OS_image_name, ValidationArgs().stg_tag('opensearch').replace(" ", ""))
+                if self.args.using_staging_artifact_only
+                else self.get_image_id(self._OS_image_name, self.args.version)
+            )
+            self.local_image_OSD_id = (
+                self.get_image_id(self._OSD_image_name, ValidationArgs().stg_tag('opensearch_dashboards').replace(" ", ""))
+                if self.args.using_staging_artifact_only
+                else self.get_image_id(self._OSD_image_name, self.args.version)
+            )
             logging.info(f'the OS image ID is : {self.local_image_OS_id}')
             logging.info(f'the OSD image ID is : {self.local_image_OSD_id} \n\n')
-
             return True
 
-        else:
-            raise Exception('Docker Daemon is not running. Exiting the docker validation.')
+        except AssertionError as e:
+            logging.error(str(e))
+            return False
+
+        finally:
+            pass
```

Comment:
Log something here instead of `pass` like what process was completed.

Reply:
got it ..it makes sense ..let me update it in next commit